src/endroid/webexhandler.py

```diff
@@ -59,42 +61,54 @@ def getOwnerList(self, room):
     def invite(self, user, room, reason):
         logging.info("Adding person %s to room %s, reason: %s", 
                      user, room, reason)
-        if self.client is not None:
+        if self.client is not None and self._is_moderator(room):
             self.client.webex_api.memberships.create(roomId=room, 
                                                      personEmail=user)
 
     def kick(self, user, room, reason): 
         logging.info("Kicking person %s from room %s, reason: %s",
                       user, room, reason)
 
-        if self.client is not None:
+        if self.client is not None and \
+           (user in self.my_emails or self._is_moderator(room)):
             memberships = self.client.webex_api.memberships.list(
                                                               roomId=room,
                                                               personEmail=user)
             for membership in memberships:
                 try:
                     self.client.webex_api.memberships.delete(
                                                     membershipId=membership.id)
-                except webexteamssdk.exceptions.ApiError as e:
-                    if e.status_code == 403:
-                        logging.error("403 error on attempt to remove user from "
-                                    "room - Endroid may not be a moderator")
-                    else:
-                        logging.exception("Got exception deleting user from "
-                                         "room")
                 except Exception:
                     logging.exception("Got exception deleting user from room")
 
     def chat(self, user, text):
         logging.info("Sending chat to user: %s", user)
+
         if self.client is not None:
-            self.client.webex_api.messages.create(toPersonEmail=user, 
-                                                  text=text) 
+            self._send_message(text=text, toPersonEmail=user)
 
     def groupChat(self, room, text):
         logging.info("Sending chat to room: %s", room)
         if self.client is not None:
-            self.client.webex_api.messages.create(roomId=room, text=text)
+            self._send_message(text=text, roomId=room)
+
+    def _send_message(self, text, **kwargs):
+        chunk = text[0:MAX_MESSAGE_LEN]
+        self.client.webex_api.messages.create(text=chunk, **kwargs)
+        if len(text) > MAX_MESSAGE_LEN:
+            reactor.callLater(0, self._send_message, text[MAX_MESSAGE_LEN:],
```

Comment:
Is there a reason not to just send all the chunks in one event (e.g. looping over chunks calling messages.create), rather than async? Just worried about whether another event could get processed in between causing another message to appear in between 2 chunks.

Reply:
I thought this was just to prevent v large messages blocking other processing. I get your point, though, will do whatever you think is best.